README.md

```diff
@@ -34,130 +49,200 @@ In this section you will start your [Google Cloud Shell](https://cloud.google.co
+$ gcloud container clusters create jenkins-cd \
   --num-nodes 3 \
-  --machine-type g1-small \
-  --scopes "https://www.googleapis.com/auth/projecthosting,https://www.googleapis.com/auth/devstorage.full_control,https://www.googleapis.com/auth/monitoring,https://www.googleapis.com/auth/logging.write,https://www.googleapis.com/auth/compute,https://www.googleapis.com/auth/cloud-platform"
+  --scopes "https://www.googleapis.com/auth/projecthosting,storage-rw"
+```
+
+Once that operation completes download the credentials for your cluster using the [gcloud CLI](https://cloud.google.com/sdk/):
+```shell
+$ gcloud container clusters get-credentials jenkins-cd
+Fetching cluster endpoint and auth data.
+kubeconfig entry generated for jenkins-cd.
 ```
 
-Now you can confirm that the cluster is running and `kubectl` is working by listing pods:
+Confirm that the cluster is running and `kubectl` is working by listing pods:
 
 ```shell
 $ kubectl get pods
 ```
+You should see an empty response.
 
-Add a quota to restrict how much of the cluster's compute and memory resources Jenkins can use:
+## Create namespace and quota for Jenkins
 
+Create the `jenkins` namespace:
 ```shell
-kubectl create -f kubernetes/jenkins/quota.yaml
+$ kubectl create ns jenkins
 ```
 
-An empty response is what you expect here.
+### Create the Jenkins Home Volume
+In order to pre-populate Jenkins with the necessary plugins and configuration for the rest of the tutorial, you will create
+a volume from an existing tarball of that data.
 
-### Create a Jenkins Replication Controller and Service
-Here you'll create a Replication Controller running a Jenkins image, and then a service that will route requests to the controller. 
+```shell
+gcloud compute images create jenkins-home-image --source-uri https://storage.googleapis.com/solutions-public-assets/jenkins-cd/jenkins-home.tar.gz
+gcloud compute disks create jenkins-home --image jenkins-home-image
+```
 
-> **Note**: All of the files that define the Kubernetes resources you will be creating for Jenkins are in the `kubernetes/jenkins` folder in the `jenkins-kube-cd` folder. You are encouraged to take a look at them before running the create commands.
+### Create a Jenkins Deployment and Service
+Here you'll create a Deployment running a Jenkins container with a persistent disk attached containing the Jenkins home directory.
 
-The Jenkins Replication Controller is defined in `kubernetes/jenkins/jenkins.yaml`. Create the controller and confirm a pod was scheduled:
+First, set the password for the default Jenkins user. Edit the password in `jenkins/k8s/options` with the password of your choice by replacing _CHANGE_ME_. To Generate a random password and replace it in the file, you can run:
 
 ```shell
-$ kubectl create -f kubernetes/jenkins/jenkins.yaml
-replicationcontrollers/jenkins-leader
+$ PASSWORD=`openssl rand -base64 15`; echo "Your password is $PASSWORD"; sed -i.bak s#CHANGE_ME#$PASSWORD# jenkins/k8s/options
```

Comment:
@viglesiasce have you considered using `envsubst` from the `gettext` package? I find it makes this kind of substitution very clean since you can use just use shell-style environment variables (e.g. `${PASSWORD}` or `$PASSWORD`) instead of sed expressions.